### ISSUE TITLE
Missing_Transcript rule misses

### ISSUE THREAD

**Issue by boonedev** [equalizedigital/accessibility-checker#165]
See: https://equalizedigital.zendesk.com/agent/tickets/5915

**Comment by amberhinds** [comment]
@boonedev @SteveJonesDev - I also saw an example of this on NASA. Their custom audio player doesn't flag missing transcript.

<img width="792" alt="Screen Shot 2023-06-15 at 7 46 14 PM" src="https://github.com/equalizedigital/accessibility-checker/assets/13153456/165f73a4-ddbd-48dc-bce9-a64410443d8c">


This is the HTML on that player:
`<div id="" class="hds-audio-player hds-module wp-block-nasa-blocks-audio-player">
<div class="hds-audio-player-wrap bg-carbon-90 padding-0 hds-audio-player--has-image" id="audioBlock-320722">
<div class="hds-audio-image">
<figure class="hds-media-background  "><img width="150" height="150" src="https://nasa-gov-preprod.go-vip.net/wp-content/uploads/2023/05/awesome-5000.jpg?w=150&amp;h=150&amp;crop=1" class="attachment-thumbnail size-thumbnail" alt="" decoding="async" loading="eager" style="transform: scale(1); transform-origin: 50% 50%; object-position: 50% 50%; object-fit: cover;" block_context="nasa-block" srcset="https://nasa-gov-preprod.go-vip.net/wp-content/uploads/2023/05/awesome-5000.jpg?resize=150,150 150w, https://nasa-gov-preprod.go-vip.net/wp-content/uploads/2023/05/awesome-5000.jpg?resize=50,50 50w, https://nasa-gov-preprod.go-vip.net/wp-content/uploads/2023/05/awesome-5000.jpg?resize=100,100 100w, https://nasa-gov-preprod.go-vip.net/wp-content/uploads/2023/05/awesome-5000.jpg?resize=200,200 200w" sizes="(max-width: 150px) 100vw, 150px"></figure>
</div>
<div class="hds-audio-player-inner padding-y-2 padding-x-3">
			<button class="hds-audio-player__playpause bg-nasa-blue-tint hds-audio-play"><br>
				<span class="playPauseIcon display-flex"><svg width="12" height="14" viewBox="0 0 12 14" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M0 14.0001V6.10352e-05L12 7.00006L0 14.0001Z" fill="white"></path></svg></span><br>
				<span class="pauseIcon display-flex" style="display: none;"><svg width="12" height="14" viewBox="0 0 12 14" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-fule="evenodd" cliprule="evenodd" d="M0 6.10352e-05H4V14.0001H0V6.10352e-05Z" fill="white"></path><path fill-fule="evenodd" cliprule="evenodd" d="M8 6.10352e-05H12V14.0001H8V6.10352e-05Z" fill="white"></path></svg></span><br>
			</button><p></p>
<div class="hds-audio-controls">
				<button class="hds-audio-player__mute hds-audio-unmuted"><br>
					<span class="volume-icon muteIcon" style="display: none;"><svg height="512" viewBox="0 0 24 24" width="512" xmlns="http://www.w3.org/2000/svg"><g><g><g><path d="m21.707 20.293-18-18c-.391-.391-1.023-.391-1.414 0s-.391 1.023 0 1.414l4.027 4.027c-.199.339-.32.77-.32 1.266v6c0 1.103.897 2 2 2h3.586l4.707 4.707c.191.191.447.293.707.293.129 0 .259-.024.383-.076.374-.154.617-.52.617-.924v-1.586l2.293 2.293c.195.195.451.293.707.293s.512-.098.707-.293c.391-.391.391-1.023 0-1.414z"></path></g><path d="m18 3v11l-7-7h.59l4.7-4.71c.29-.28.72-.37 1.09-.21.38.15.62.52.62.92z"></path></g></g></svg></span><br>
					<span class="volume-icon volumeIcon"><svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlnsxlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 512 512" style="enableBackground: new 0 0 512 512" xmlspace="preserve"><g><g><path d="M85.2,163.8h82.2c7.5,0,13.5,6.1,13.5,13.5v170.9c0,7.5-6.1,13.5-13.5,13.5H85.2c-7.5,0-13.5-6.1-13.5-13.5 V177.4C71.7,169.9,77.7,163.8,85.2,163.8z"></path></g></g><g><g><path d="M215,170.3V355c0,4.3,2.1,8.4,5.5,10.9l177.7,130.3c8.9,6.6,21.5,0.2,21.5-10.9V28.1 c0-11.3-13.1-17.6-22-10.6L220.1,159.8C216.9,162.3,215,166.2,215,170.3z"></path></g></g></svg></span><br>
				</button><p></p>
<div class="hds-audio-player__volume-wrapper">
					<input type="range" min="0" max="1" step="0.01" value="1" class="hds-audio-player__volume">
				</div>
<p></p></div>
<div class="hds-audio-player__scrubber-time">
				<input type="range" min="0" max="100" value="0" step="1" class="hds-audio-player__scrubber"><p></p>
<p class="hds-audio-time line-height-sm color-carbon-40 paddding-0 margin-0">0:00 / 0:0</p>
<p></p></div>
<p></p></div>
<p>		<audio controls="" class="audioPlayer" style="display: none;"><source src="https://nasa-gov-preprod.go-vip.net/wp-content/uploads/2020/10/ga_david_smith_mix_2.mp3" type="audio/mpeg">Your browser does not support the audio element.</audio>
	</p></div>
<p></p></div>`

https://nasa-gov-preprod.go-vip.net/general/audio-player-test/ (Only Steve will be able to access this URL because of the IP restrictions.)

**Comment by boonedev** [comment]
@SteveJonesDev @amberhinds I think this may be the issue: 
https://github.com/equalizedigital/accessibility-checker/blob/e7db78466a71c5b80b432d637cd929d6752a69d3/includes/rules/missing_transcript.php#L21

`$post->post_content` is pulling directly from the db, and if the content is block-based then that field holds the block config info rather than the final html ie:

```
<!-- wp:embed {"url":"https://soundcloud.com/djspoonsantos/vibe-original-mix-2k21?utm_source=clipboard\u0026utm_medium=text\u0026utm_campaign=social_sharing","type":"rich","providerNameSlug":"soundcloud","responsive":true} -->
<figure class="wp-block-embed is-type-rich is-provider-soundcloud wp-block-embed-soundcloud"><div class="wp-block-embed__wrapper">
https://soundcloud.com/djspoonsantos/vibe-original-mix-2k21?utm_source=clipboard&amp;utm_medium=text&amp;utm_campaign=social_sharing
</div></figure>
<!-- /wp:embed -->

```

So when the rule looks at the Soundcloud block for "iframe", "video", "audio" or ".is-type-video" as a trigger to search for "transcript", it doesn't find it. If the `post_content` includes one of the video blocks (ie. YouTube), that config includes a `"type":"video"`, so it will run the search for "transcript".

I think instead if we swap it to use the content pulled from the rendered page (ie.`$content['html']->save()`) then WP will have gone thru the block rendering process to produce the final html for the block, so the rule will find "iframe" and run:
```
<figure class="wp-block-embed is-type-rich is-provider-soundcloud wp-block-embed-soundcloud"><div class="wp-block-embed__wrapper">
<iframe loading="lazy" title="Vibe (Original Mix) 2K21 by Dj Spoon" width="500" height="400" scrolling="no" frameborder="no" src="https://w.soundcloud.com/player/?visual=true&#038;url=https%3A%2F%2Fapi.soundcloud.com%2Ftracks%2F1111259281&#038;show_artwork=true&#038;maxheight=750&#038;maxwidth=500"></iframe>
</div></figure>
```

Let me know if this is the route we want to go and I'll put together a PR for it and include missing_headings and incorrect_heading_order which also use `$post->post_content` (I didn't find any others when I ran a grep).

**Comment by SteveJonesDev** [comment]
Closing a stale. This rule has been completely rewritten in JavaScript.
